strawberry/extensions/tracing/opentracing.py

```diff
@@ -0,0 +1,237 @@
+        self._root_scope.span.set_tag(tags.COMPONENT, "graphql")
+
+    def on_request_end(self, *, execution_context: ExecutionContext):
+        self._root_scope.close()
+        self.end_timestamp = self.now()
+        self.end_time = datetime.utcnow()
+
+    def on_parsing_start(self):
+        self._start_parsing = self.now()
+
+    def on_parsing_end(self):
+        self._end_parsing = self.now()
+
+    def on_validation_start(self):
+        self._start_validation = self.now()
+
+    def on_validation_end(self):
+        self._end_validation = self.now()
+
+    def now(self) -> int:
+        return time.perf_counter_ns()
+
+    @property
+    def stats(self) -> OpenTracingStats:
+        return OpenTracingStats(
+            start_time=self.start_time,
+            end_time=self.end_time,
+            duration=self.end_timestamp - self.start_timestamp,
+            execution=OpenTracingExecutionStats(self._resolver_stats),
+            validation=OpenTracingStepStats(
+                start_offset=self._start_validation - self.start_timestamp,
+                duration=self._end_validation - self._start_validation,
+            ),
+            parsing=OpenTracingStepStats(
+                start_offset=self._start_parsing - self.start_timestamp,
+                duration=self._end_parsing - self._start_parsing,
+            ),
+        )
+
+    def get_results(self):
+        return {"tracing": self.stats.to_json()}
+
+    def filter_resolver_args(
+        self, args: Dict[str, Any], info: GraphQLResolveInfo
+    ) -> Dict[str, Any]:
+        if not self._arg_filter:
+            return args
+        return self._arg_filter(deepcopy(args), info)
+
+    async def resolve(self, _next, root, info, *args, **kwargs):
+        start_timestamp = self.now()
+
+        resolver_stats = OpenTracingResolverStats(
+            path=get_path_from_info(info),
+            field_name=info.field_name,
+            parent_type=info.parent_type,
+            return_type=info.return_type,
+            start_offset=start_timestamp - self.start_timestamp,
+        )
+
+        try:
+            if not do_trace(info):
+                result = _next(root, info, *args, **kwargs)
+                if isawaitable(result):
+                    result = await result
+                return result
+
+            with self._tracer.start_active_span(info.field_name) as scope:
+                span = scope.span
+                span.set_tag(tags.COMPONENT, "graphql")
+                span.set_tag("graphql.parentType", info.parent_type.name)
+
+                graphql_path = ".".join(map(str, get_path_from_info(info)))
+                span.set_tag("graphql.path", graphql_path)
+                if kwargs:
+                    filtered_kwargs = self.filter_resolver_args(kwargs, info)
+                    for kwarg, value in filtered_kwargs.items():
+                        span.set_tag(f"graphql.param.{kwarg}", value)
```

Comment:
since we have this in the sync code, shall we move it to a function and reuse it? :)